client/src/pages/OnboardingPage/PersonalSettingsForm/PersonalSettingsForm.js

```diff
@@ -16,16 +16,59 @@ class PersonalSettingsForm extends Component {
         dream_destination: "",
         first_name: this.props.setup.first_name,
         last_name: this.props.setup.last_name,
-        city_of_residence: this.props.filter.city
-      }
+        city_of_residence: this.props.filter.city,
+      },
     };
   }
+  getSignedRequest = (file) => {
+    // copied and modified from https://devcenter.heroku.com/articles/s3-upload-python
+    let xhr = new XMLHttpRequest();
+    // this function/method should only be called in prod, so heroku URL is used directly
+    xhr.open(
+      "GET",
+      "https://cotripper-api.herokuapp.com/sign_s3?file_name=" +
+        file.name +
+        "&file_type=" +
+        file.type
+    );
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === 4) {
+        if (xhr.status === 200) {
+          var response = JSON.parse(xhr.responseText);
+          this.uploadFile(file, response.data, response.url);
+        } else {
+          alert("Could not get signed URL.");
+        }
+      }
+    };
+    xhr.send();
+  };
+  uploadFile = (file, s3Data, url) => {
+    let xhr = new XMLHttpRequest();
+    xhr.open("POST", s3Data.url);
 
-  updateValue = e => {
+    let postData = new FormData();
+    for (let key in s3Data.fields) {
+      postData.append(key, s3Data.fields[key]);
+    }
+    postData.append("file", file);
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === 4) {
+        if (xhr.status === 200 || xhr.status === 204) {
+          this.setState({ profile: { ...this.state.profile, image: url } });
+        } else {
+          alert("Could not upload file.");
+        }
+      }
+    };
+    xhr.send(postData);
```

Comment:
Can you update this to use `fetch` instead of `XMLHttpRequest`? The former is a much cleaner, more modern way of handling requests. Alternatively, look into using [Axios](https://github.com/axios/axios).

Reply:
Thank you again for the help with axios! Should be switched over now.